src/usr/local/www/diag_backup.php

```diff
@@ -121,26 +121,21 @@ function restore_rrddata() {
 	}
 }
 
-function restore_captiveportal_data($type='voucher') {
+function restore_xmldatafile($type='voucher') {
 	global $config, $g;
-	if ($type == 'captiveportal_usedmacs') {
-		$conf = 'captiveportal';
-	} else {
-		$conf = $type;
-	}
-	foreach ($config[$conf]["{$type}data"]["{$type}dbfile"] as $db) {
-		$cpdata_file = "{$g['vardb_path']}/{$db['filename']}";
-		if (file_put_contents($cpdata_file, gzinflate(base64_decode($db['dbdata']))) === false) {
-			log_error(sprintf(gettext("Cannot write %s"), $cpdata_file));
+	foreach ($config[$type]["{$type}data"]["xmldatafile"] as $file) {
+		$xmldata_file = "{$g['vardb_path']}/{$file['filename']}";
```

Comment:
For safety, this should use `basename()` on the contents of the filename to ensure it doesn't write to the wrong place accidentally (e.g. someone manually edits their config wrong). Otherwise it could have relative path references, subdirectories, etc, which doesn't appear to be necessary or intended.

Reply:
Thanks for info, fixed